File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        newArr = []
        for i in range(len(arr)):
            max = -1
            # find Max
            for j in range(i+1, len(arr)):
                num = arr[j]
                if num > max:
                    max = num 
            
            newArr.append(max)
            
        return newArr

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int maxArea = 0;
        int l = 0;
        int r = heights.length - 1;

        while (l < r) {
            if (heights[l] > heights[r]) {
                maxArea = Math.max(maxArea, (r - l) * heights[r]);
                r--;
            } else {
                maxArea = Math.max(maxArea, (r - l) * heights[l]);
                l++;
            }
        }
        return maxArea;
    }
}
